x-pack/test/api_integration/deployment_agnostic/apis/observability/apm/observability_overview/has_data.spec.ts

```diff
@@ -7,51 +7,57 @@
 
 import expect from '@kbn/expect';
 
-import { FtrProviderContext } from '../../common/ftr_provider_context';
+import type { DeploymentAgnosticFtrProviderContext } from '../../../../ftr_provider_context';
+import { ARCHIVER_ROUTES } from '../constants/archiver';
 
-export default function ApiTest({ getService }: FtrProviderContext) {
-  const registry = getService('registry');
-  const apmApiClient = getService('apmApiClient');
+export default function ApiTest({ getService }: DeploymentAgnosticFtrProviderContext) {
+  const apmApiClient = getService('apmApi');
+  const esArchiver = getService('esArchiver');
 
-  registry.when(
-    'Observability overview when data is not loaded',
-    { config: 'basic', archives: [] },
-    () => {
+  describe('has data', () => {
+    describe('when no data is loaded', () => {
       it('returns false when there is no data', async () => {
         const response = await apmApiClient.readUser({
           endpoint: 'GET /internal/apm/observability_overview/has_data',
         });
         expect(response.status).to.be(200);
         expect(response.body.hasData).to.eql(false);
       });
-    }
-  );
+    });
+
+    describe('when only onboarding data is loaded', () => {
+      before(async () => {
+        await esArchiver.load(ARCHIVER_ROUTES.observability_overview);
+      });
+
+      after(async () => {
+        await esArchiver.unload(ARCHIVER_ROUTES.observability_overview);
+      });
 
-  registry.when(
-    'Observability overview when only onboarding data is loaded',
-    { config: 'basic', archives: ['observability_overview'] },
-    () => {
       it('returns false when there is only onboarding data', async () => {
         const response = await apmApiClient.readUser({
           endpoint: 'GET /internal/apm/observability_overview/has_data',
         });
         expect(response.status).to.be(200);
         expect(response.body.hasData).to.eql(false);
       });
-    }
-  );
+    });
+
+    describe('when data is loaded', () => {
+      before(async () => {
+        await esArchiver.load(ARCHIVER_ROUTES['8.0.0']);
```

Comment:
Using `8.0.0` as `apm_8.0.0` contains `.ml-config` indices which make MKI unable to pass. The logic stills the same as this endpoint only checks if there is any apm data, and there is.

Reply:
Thanks for mentioning this! I’ll double-check a few migration issues I previously discarded for similar reasons to see if that change might allow them to work without the ML index on the mappings.